File: ng/src/app/models/expense.ts
import { Category } from '@/models/category';

export class Expense {
    constructor() {
        this.date = new Date();
        this.id = 0;
        this.description = '';
        this.categoryId = 0;
        this.registeredBy = '';
        this.familyId = 0;
    }

    id: number;
    amount?: number;
    description: string;
    category?: Category;
    categoryId: number;
    registeredBy: string;
    date: Date;
    familyId: number;
}

File: api/mathew/entities/Expense.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using mathew.entities;

namespace mathew.entities;
public class Expense
{
    public int Id { get; set; }

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal Amount { get; set; }

    [Required]
    [MaxLength(500)]
    public string Description { get; set; } = string.Empty;

    [Required]
    public DateTimeOffset Date { get; set; }

    [Required]
    public int CategoryId { get; set; }

    public Category? Category { get; set; } = null!;
    [Required]
    public string RegisteredBy { get; set; }


}